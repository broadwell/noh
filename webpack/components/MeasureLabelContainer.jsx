import React from "react";
import PropTypes from "prop-types";

const MeasureLabelContainer = props => {
  let labels = "";
  const { isBeatOn, isTextOn, isPercussionOn, isNohkanOn, isDanceOn } = props;
  const hasSomeToggleOn = Object.values({
    isBeatOn,
    isTextOn,
    isPercussionOn,
    isNohkanOn,
    isDanceOn
  }).some(value => !!value);
  if (props.previous === true && hasSomeToggleOn) {
    labels = <div className="measure__label-container">Previous sentence</div>;
  } else if (props.next === true && hasSomeToggleOn) {
    labels = <div className="measure__label-container">Next sentence</div>;
  } else {
    const measures = [
      isBeatOn ? (
        <div
          key="beatLabel"
          className="measure__channel measure__channel--beat"
        >
          Beat
        </div>
      ) : (
        ""
      ),
      isTextOn ? (
        <div
          key="textLabel"
          className="measure__channel measure__channel--text measure__channel--large"
        >
          Text
        </div>
      ) : (
        ""
      ),
      isPercussionOn ? (
        <div
          key="percussionLabel"
          className="measure__channel measure__channel--percussion"
        >
          Percussion
        </div>
      ) : (
        ""
      ),
      isNohkanOn ? (
        <div
          key="nohkanLabel"
          className="measure__channel measure__channel--nohkan"
        >
          Nohkan
        </div>
      ) : (
        ""
      ),
      isDanceOn ? (
        <div
          key="danceLabel"
          className="measure__channel measure__channel--dance"
        >
          Dance
        </div>
      ) : (
        ""
      )
    ];
    labels = <div className="measure__label-container">{measures}</div>;
  }
  return labels;
};

MeasureLabelContainer.PropTypes = {
  previous: PropTypes.bool,
  next: PropTypes.bool,
  isBeatOn: PropTypes.bool,
  isTextOn: PropTypes.bool,
  isPercussionOn: PropTypes.bool,
  isNohkanOn: PropTypes.bool,
  isDanceOn: PropTypes.bool,
  isPrevSentenceOn: PropTypes.bool,
  isNextSentenceOn: PropTypes.bool
};

MeasureLabelContainer.defaultProps = {
  previous: false,
  next: false,
  isBeatOn: true,
  isTextOn: true,
  isPercussionOn: true,
  isNohkanOn: true,
  isDanceOn: true,
  isPrevSentenceOn: true,
  isNextSentenceOn: true
};

export default MeasureLabelContainer;
