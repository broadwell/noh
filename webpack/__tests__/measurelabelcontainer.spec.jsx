import React from "react";
import { shallow } from "enzyme";
import MeasureLabelContainer from "../components/MeasureLabelContainer";

describe("<MeasureLabelContainer>", () => {
  it("renders as expected by default", () => {
    const component = shallow(<MeasureLabelContainer />);
    expect(component).toMatchSnapshot();
  });
  it("renders as expected for previous", () => {
    const component = shallow(<MeasureLabelContainer previous />);
    expect(component).toMatchSnapshot();
  });
  it("renders as expected for next", () => {
    const component = shallow(<MeasureLabelContainer next />);
    expect(component).toMatchSnapshot();
  });
});
