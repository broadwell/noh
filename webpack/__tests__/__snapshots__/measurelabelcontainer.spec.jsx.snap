// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MeasureLabelContainer> renders as expected by default 1`] = `
<div
  className="measure__label-container"
>
  <div
    className="measure__channel measure__channel--beat"
  >
    Beat
  </div>
  <div
    className="measure__channel measure__channel--text measure__channel--large"
  >
    Text
  </div>
  <div
    className="measure__channel measure__channel--percussion"
  >
    Percussion
  </div>
  <div
    className="measure__channel measure__channel--nohkan"
  >
    Nohkan
  </div>
  <div
    className="measure__channel measure__channel--dance"
  >
    Dance
  </div>
</div>
`;

exports[`<MeasureLabelContainer> renders as expected for next 1`] = `
<div
  className="measure__label-container"
>
  Next sentence
</div>
`;

exports[`<MeasureLabelContainer> renders as expected for previous 1`] = `
<div
  className="measure__label-container"
>
  Previous sentence
</div>
`;
